packages/aws-cdk/test/api/hotswap-deployments.test.ts

```diff
@@ -1,343 +1,185 @@
-  expect(mockUpdateLambdaCode).toHaveBeenCalledWith({
-    FunctionName: 'lambda-mybucket-function',
-    S3Bucket: 'current-bucket',
-    S3Key: 'new-key',
-  });
+  expect(deployStackResult?.noOp).toEqual(true);
+  expect(mockUpdateMachineDefinition).not.toHaveBeenCalled();
+  expect(mockUpdateLambdaCode).not.toHaveBeenCalled();
 });
 
-test("correctly falls back to taking the function's name from the current stack if it can't evaluate it in the template", async () => {
+test('resource deletions require full deployments', async () => {
   // GIVEN
-  currentCfnStack.setTemplate({
-    Parameters: {
-      Param1: { Type: 'String' },
-      AssetBucketParam: { Type: 'String' },
-    },
+  setup.currentCfnStack.setTemplate({
     Resources: {
-      Func: {
-        Type: 'AWS::Lambda::Function',
+      Machine: {
+        Type: 'AWS::StepFunctions::StateMachine',
         Properties: {
-          Code: {
-            S3Bucket: { Ref: 'AssetBucketParam' },
-            S3Key: 'current-key',
+          DefinitionString: {
+            'Fn::Join': [
+              '',
+              [
+                '{ "Prop" : "old-value", "AnotherProp" : "another-old-value" }',
+              ],
+            ],
           },
-          FunctionName: { Ref: 'Param1' },
-        },
-        Metadata: {
-          'aws:asset:path': 'old-path',
+          StateMachineName: 'my-machine',
         },
       },
     },
   });
-  currentCfnStackResources.push(stackSummaryOf('Func', 'AWS::Lambda::Function', 'my-function'));
-  const cdkStackArtifact = cdkStackArtifactOf({
+  const cdkStackArtifact = setup.cdkStackArtifactOf({
     template: {
-      Parameters: {
-        Param1: { Type: 'String' },
-        AssetBucketParam: { Type: 'String' },
-      },
       Resources: {
-        Func: {
-          Type: 'AWS::Lambda::Function',
-          Properties: {
-            Code: {
-              S3Bucket: { Ref: 'AssetBucketParam' },
-              S3Key: 'new-key',
-            },
-            FunctionName: { Ref: 'Param1' },
-          },
-          Metadata: {
-            'aws:asset:path': 'new-path',
-          },
-        },
       },
     },
   });
 
   // WHEN
-  const deployStackResult = await tryHotswapDeployment(mockSdkProvider, {
-    AssetBucketParam: 'asset-bucket',
-  }, currentCfnStack, cdkStackArtifact);
+  const deployStackResult = await tryHotswapDeployment(setup.mockSdkProvider, {}, setup.currentCfnStack, cdkStackArtifact);
 
   // THEN
-  expect(deployStackResult).not.toBeUndefined();
-  expect(mockUpdateLambdaCode).toHaveBeenCalledWith({
-    FunctionName: 'my-function',
-    S3Bucket: 'asset-bucket',
-    S3Key: 'new-key',
-  });
+  expect(deployStackResult).toBeUndefined();
+  expect(mockUpdateLambdaCode).not.toHaveBeenCalled();
+  expect(mockUpdateMachineDefinition).not.toHaveBeenCalled();
 });
 
-test("will not perform a hotswap deployment if it cannot find a Ref target (outside the function's name)", async () => {
```

Comment:
This test has been lost, right? (It's been deleted from here, and it's not in `lambda-hotswap-deployments.test.ts`).

Can we bring it back please?